src/sacn/private/dmx_merger.h

```diff
@@ -29,14 +29,63 @@
 #include <stdbool.h>
 #include <stdint.h>
 #include "../../../include/sacn/dmx_merger.h"
+#include "sacn/private/util.h"
+#include "etcpal/rbtree.h"
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+typedef struct SourceState
+{
+  sacn_source_id_t handle;
+  SacnDmxMergerSource source;
+} SourceState;
+
+typedef struct CidToSourceHandle
+{
+  EtcPalUuid cid;
+  sacn_source_id_t handle;
+} CidToSourceHandle;
+
+typedef struct MergerState
+{
+  sacn_dmx_merger_t handle;
+
+  IntHandleManager source_handle_mgr;
+
+  EtcPalRbTree source_state_lookup;
+  EtcPalRbTree source_handle_lookup;
+
+  const SacnDmxMergerConfig* config;
+  uint8_t winning_priorities[DMX_ADDRESS_COUNT];
+} MergerState;
+
+IntHandleManager merger_handle_mgr;
+EtcPalRbTree mergers;
+
 etcpal_error_t sacn_dmx_merger_init();
 void sacn_dmx_merger_deinit(void);
 
+int merger_state_lookup_compare_func(const EtcPalRbTree* self, const void* value_a, const void* value_b);
```

Comment:
In general, functions used only inside the .c file should be defined as static at the top of the .c file, not in the private header.